LayoutTests/fast/dom/empty-hash-and-search.html

```diff
@@ -32,9 +32,6 @@
         window.location += "?#";
 
 } else { // Added "?#" and reloaded.
-    var link1 = document.getElementById("link1");
```

Comment:
Why are these changes here?

Reply:
Discussed in the commit message. The test is broken, for the case where the test fails.

Reply:
I read the commit message, and I still don't understand why this change is correct. If anything, it's the above block that is wrong, as it uses undefined variables?